presto-main/src/test/java/com/facebook/presto/sql/planner/assertions/SymbolMatcher.java

```diff
@@ -40,8 +38,8 @@ public boolean matches(PlanNode node, Session session, Metadata metadata, Expres
     {
         Symbol symbol = null;
         for (Symbol outputSymbol : node.getOutputSymbols()) {
-            if (pattern.matcher(outputSymbol.getName()).find()) {
-                checkState(symbol == null, "%s symbol was found multiple times in %s", pattern, node.getOutputSymbols());
+            if (expectedSymbol.equals(outputSymbol)) {
```

Comment:
I generally don't like the idea of `StemSymbol`. If I undestand it correctly its motivation is to handle the case that planner may be not deterministic and symbols have dynamic name suffixes which are unknown before the execution.

I thought I solved this issue with `ExpressionAliases` (previously it was `SymbolAliases`, but `Expression` is more general as it can handle literals etc.). It can register an expression in one part of the plan and then and check the usage in other part of the plan.

To see example usage take a look at:
 - `com.facebook.presto.sql.planner.TestLogicalPlanner#testJoin`  - notice there are two symbols named `orderkey`. They have different suffixes in plan. Their aliases are checked in `JoinMatcher` to see that they are used as `EquiCriteria` for inner join.
 - com.facebook.presto.sql.planner.TestLogicalPlanner#testCorrelatedSubqueries (2nd test) - see that `custkey` is used in filter expression. There are also two `orderkey` symbols.

We can call and talk about it (ping me on slack).